src/server.js

```diff
@@ -127,16 +132,11 @@ module.exports = function (classes) {
         var server = http.createServer();
 
         server.on('request', function onRequest(req, res) {
-          self.handleHttp(req, res)
-          .then(function(){
-          })
-          .catch(Error.ParseError, function(err){
-            Server.handleHttpError(req, res, err, self.opts.headers);
-          })
-          .catch(Error.InvalidRequest, function(err){
-            Server.handleHttpError(req, res, err, self.opts.headers);
-          })
-          ;
+          self.handleHttp(req, res).then(function () {
+          }).catch(function (err) {
```

Comment:
the type of errors must be caught using bluebird `catch`, because if a TypeError, ReferenceError happens, it will be caught and swalled by the handleHttpError, and we don't want to give the client internal errors for security reasons. http://bluebirdjs.com/docs/api/catch.html it's like a try / catch, but for promises